addons/ekco/0.26.1/install.sh

```diff
@@ -342,6 +342,15 @@ function ekco_bootstrap_internal_lb() {
     if [ "$MASTER" = "1" ] && commandExists "kubectl" ; then
         kubectl -n kurl delete --ignore-not-found configmap update-internallb &>/dev/null || true
     fi
+
+    # Ensure we can read this file even when there is a system wide umask policy that prohibits
+    # reading the file such as 'umask 0027'.
+    # The kubelet creates a static haproxy pod which mounts haproxy.cfg as a hostpath volume. If
+    # the file does not have read permission for 'others', the haproxy container will not start
+    # during kubeadm init.
+    if [ -f /etc/haproxy/haproxy.cfg ]; then
+        chmod -R o+rx /etc/haproxy
```

Comment:
We ( as in kURL) own this file so I think it's perfectly fine for us to adjust the permissions so it can work with our installer.

Reply:
What do you mean with we own the file? Have we any doc/guidance that says that kURL can change the permissions of any file created in the system that it requires for it work without the admins knowledge and consent?

IMO we need to raise an an error in this scenario, we need to have document that clarifies what are the permissions and requirements for kURL properly work and we should not change permissions of files in the system that the admin explicit decide that should be less privileged without inform and make ask for their consent. 

In POV it is not a right approach to deal with this kind of scenario.

Reply:
I think @chris-sanders worked in the analyse of this one right? Let's ask a hand on this one before move forward.

Reply:
We own the file we should be setting permissions as needed. 
I am curious why we're setting +x on a config file, shouldn't this be o+rw?

Reply:
We "own" the file that seems very strong. we created the file and we need to use it, but someone decide to explicit change the permissions afterwords. Som would not make more sense say "OK you changed the permissions here and we need this one to properly work, are you OK with?"

However, if all is OK with that is fine. Could we at least log, warning and notify that happened?
For example, could we raise an event for this one and log the change? 

+1 for the question over why we need +x in this one?

Reply:
I'm setting `+x` permission because `/etc/haproxy` is a directory.

Reply:
Writing to this directory is not a problem since the the kURL installer script runs as root. Also, when we write the haproxy manifest, we mount the /etc/haproxy dir  with read+write permission:https://github.com/replicatedhq/kURL/blob/fb6777b4a6a87914b70dbcb34a02baa6d5ca0a67/addons/ekco/template/base/install.sh#L270-L281

Reply:
Ah ok then what about using `chomd -R o=rwX` which will set the directory to +x and the files in it to `rw`, I'm assuming you're doing this after the files are created and want to change both the folder and it's contents?

Reply:
To follow up on

> OK you changed the permissions here 

This isn't happening though. They aren't changing the permissions the file simply has these permissions by default. The file didn't exist until we created it, and we can/should create it with appropriate permissions. However, even if they did change it after the fact that brings me to the other point.

> We "own" the file that seems very strong

You are right it's a strong statement, but strong doesn't mean it's wrong. We do in fact own this file, it's for our service, we can re-write it at any point and would never expect user edits to be persisted, this includes both the contents of the file and it's access permissions. This file for all purposes is part of kURL and users should never touch it, it's ours.

Reply:
>Ah ok then what about using chomd -R o+rwX which will set the directory to +x and the files in it to rw, I'm assuming you're doing this after the files are created and want to change both the folder and it's contents?

@chris-sanders I'm hesitant in granting write access to the world/other group for kURL created files so I used the minimum perms needed that makes us work. If we encounter a scenario where we need to tweak permissions again, I'll change it.